Zend/zend_language_parser.y

```diff
@@ -1077,15 +1076,21 @@ property:
 
 class_const_list:
 		class_const_list ',' class_const_decl { $$ = zend_ast_list_add($1, $3); }
-	|	class_const_decl { $$ = zend_ast_create_list(1, ZEND_AST_CLASS_CONST_DECL, $1); }
+	|	first_class_const_decl { $$ = zend_ast_create_list(1, ZEND_AST_CLASS_CONST_DECL, $1); }
+;
+
+first_class_const_decl:
+                T_STRING '=' expr backup_doc_comment { $$ = zend_ast_create(ZEND_AST_CONST_ELEM, $1, $3, ($4 ? zend_ast_create_zval_from_str($4) : NULL), NULL); }
+        |       semi_reserved '=' expr backup_doc_comment { zval zv; if (zend_lex_tstring(&zv, $1) == FAILURE) { YYABORT; } $$ = zend_ast_create(ZEND_AST_CONST_ELEM, zend_ast_create_zval(&zv), $3, ($4 ? zend_ast_create_zval_from_str($4) : NULL), NULL); }
+        |       type_expr identifier '=' expr backup_doc_comment { $$ = zend_ast_create(ZEND_AST_CONST_ELEM, $2, $4, ($5 ? zend_ast_create_zval_from_str($5) : NULL), $1); }
```

Comment:
This isn't right: The type should be part of `CLASS_CONST_DECL`, not `CONST_ELEM`, and apply to all constants in the declaration.

Consider something like this:
```
class Test {
    const int X = 1, Y = "foo";
}
```
Here the `int` type should apply to both `X` and `Y`, while it currently applies only to `X`.

Reply:
Correction: It should be on CLASS_CONST_GROUP.

Reply:
I've put up https://github.com/php/php-src/pull/11286 to fix.

Reply:
Thank you for spotting and fixing this! Applying the type to all constant elements was my intention, but I forgot to take care of the type of the rest of the constants during compilation... But I'm glad that you managed to find a more elegant solution for circumventing the parser issues :)